lib/easybill-rest-client/generic_api.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+module EasybillRestClient
+  class GenericApi
```

Comment:
Why don't we make it a regular class with class-level attribute readers for `resource_name` and `resource_class`, instead? Wouldn't that be much simpler?